sbi/inference/posteriors/base_posterior.py

```diff
@@ -582,6 +591,139 @@ def sample_conditional(
 
         return samples.reshape((*sample_shape, -1))
 
+    def map(
+        self,
+        x: Optional[Tensor] = None,
+        num_iter: int = 1000,
+        learning_rate: float = 0.1,
+        init_method: Union[str, Tensor] = "posterior",
+        num_init_samples: int = 1_000,
+        num_to_optimize: int = 100,
+        show_progress_bars: bool = True,
+        log_prob_kwargs: Dict = {},
+    ) -> Tensor:
+        """
+        Returns the maximum-a-posteriori estimate (MAP).
+
+        The MAP is obtained by running gradient ascent from a given number of starting
+        positions (samples from the posterior with the highest log-probability). After
+        the optimization is done, we select the parameter set that has the highest
+        log-probability after the optimization.
```

Comment:
repetition in the sentence.

Reply:
Not sure how else to formulate it. We indeed check for the highest log-prob twice: once before the optimization (to find good init location) and once after the optimization (to find the best sample).